oximeter/impl/src/schema/mod.rs

```diff
@@ -297,6 +297,24 @@ impl TimeseriesSchema {
         self.field_schema.iter().find(|field| field.name == name.as_ref())
     }
 
+    /// Return an iterator over the target fields.
+    pub fn target_fields(&self) -> impl Iterator<Item = &FieldSchema> {
+        self.field_iter(FieldSource::Target)
+    }
+
+    /// Return an iterator over the metric fields.
+    pub fn metric_fields(&self) -> impl Iterator<Item = &FieldSchema> {
+        self.field_iter(FieldSource::Metric)
+    }
+
+    /// Return an iterator over fields from the given source.
+    fn field_iter(
+        &self,
+        source: FieldSource,
+    ) -> impl Iterator<Item = &FieldSchema> {
+        self.field_schema.iter().filter(move |field| field.source == source)
```

Comment:
I'm a little surprised `move` is necessary here, `FieldSource` already implements `Copy`

Reply:
I am as well, actually. I didn't dig too much into it. I'll do that now, since it seems like we're both missing something (or it's a rustc bug!).

Reply:
Ok, here's the error:

```
bnaecker@shale : ~/omicron/oximeter/impl $ cargo c
    Checking oximeter-impl v0.1.0 (/home/bnaecker/omicron/oximeter/impl)
error[E0373]: closure may outlive the current function, but it borrows `source`, which is owned by the current function
   --> oximeter/impl/src/schema/mod.rs:322:41
    |
322 |         self.field_schema.iter().filter(|field| field.source == source)
    |                                         ^^^^^^^                 ------ `source` is borrowed here
    |                                         |
    |                                         may outlive borrowed value `source`
    |
note: closure is returned here
   --> oximeter/impl/src/schema/mod.rs:322:9
    |
322 |         self.field_schema.iter().filter(|field| field.source == source)
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
help: to force the closure to take ownership of `source` (and any other referenced variables), use the `move` keyword
    |
322 |         self.field_schema.iter().filter(move |field| field.source == source)
    |                                         ++++

For more information about this error, try `rustc --explain E0373`.
error: could not compile `oximeter-impl` (lib) due to 1 previous error
```

That seems legit. The closure owns the reference in `field`, but borrows `source`. So when the function `field_iter()` returns, `source` is dropped, but the closure lives on!